### PR TITLE
Implement an internal mapcache

### PR DESCRIPTION
**Pull request by pvalsecc**
Also switch to circleCI

### CODE REVIEW

File: tilecloud_chain/__init__.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def _resolution_scale(self, resolutions):
 
     @staticmethod
     def _configure_logging(options, format_):
-        if os.environ.get('CI', 'FALSE') == 'TRUE':
+        if os.environ.get('CI', 'TRUE') == 'TRUE':
```
Comment (sbrunner) [inline tilecloud_chain/__init__.py]: Looks wrong ...

File: tilecloud_chain/generate.py [hunk 2]
```diff
@@ -379,7 +377,7 @@ def _get_tilestore_for_layer(self, layer):
                 from tilecloud.store.mapnik_ import MapnikTileStore
                 from tilecloud_chain.mapnik_ import MapnikDropActionTileStore
             except ImportError:
-                if 'CI' not in os.environ:  # pragma nocover
+                if os.environ.get('CI', 'TRUE') == 'FALSE':  # pragma nocover
```
Comment (sbrunner) [inline tilecloud_chain/generate.py]: Same ...

File: tilecloud_chain/tests/test_error.py [hunk 3]
```diff
@@ -151,7 +151,6 @@ def test_wrong_srs(self, l):
 
     @log_capture('tilecloud_chain')
     @attr(general=True)
-    @attr(nopy2=True)
     @attr(nopy3=True)
```
Comment (sbrunner) [inline tilecloud_chain/tests/test_error.py]: For me this test should work on py3 or be removed ...

File: tilecloud_chain/tests/test_error.py [hunk 4]
```diff
@@ -180,7 +178,7 @@ def test_wrong_sequence(self, l):
             'tilecloud_chain', 'ERROR', """The config file 'tilegeneration/wrong_sequence.yaml' is invalid.
  - /: Cannot find required key 'caches'.
  - /: Cannot find required key 'layers'.
- - /grids/test/resolutions: Value 'test' is not a list. Value path: '/grids/test/resolutions'
+ - /grids/test/resolutions: Value 'b'test'' is not a list. Value path: '/grids/test/resolutions'
```
Comment (sbrunner) [inline tilecloud_chain/tests/test_error.py]: :-/

File: tilecloud_chain/__init__.py (hunk 1 above)
Comment (pvalsecc) [reply tilecloud_chain/__init__.py]: Indeed! I've missed the fact this is production code...

File: tilecloud_chain/tests/test_error.py (hunk 4 above)
Comment (pvalsecc) [reply tilecloud_chain/tests/test_error.py]: This message is not built by us... So I cannot fix it.

**Pull request merged**